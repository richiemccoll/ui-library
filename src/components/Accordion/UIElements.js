import styled from 'styled-components';

export const Container = styled.ul`
  width: 90%;
  position: relative;
  background-color: #f2f2f2;
  padding: 16px;
  transition: .2s background-color linear;
  list-style: none;
  box-shadow: 1px 1px 1px 1px #f2f2f2;

  &:hover {
    background-color: #ccc;
    cursor: pointer;
  }
`;

export const IconContainer = styled.span`
  position: absolute;
  right: 0;
  margin-right: 16px;
`;