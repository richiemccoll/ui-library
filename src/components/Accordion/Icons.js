import React from 'react';
import { IconContainer } from './UIElements';

export const Icons = ({ isOpen }) => {
  return (
    <IconContainer>
      {isOpen
        ? <svg
            id="i-chevron-top"
            viewBox="0 0 32 32"
            width="16"
            height="16"
            fill="none"
            stroke="currentcolor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          >
            <path d="M30 20 L16 8 2 20" />
          </svg>
        : <svg
            id="i-chevron-bottom"
            viewBox="0 0 32 32"
            width="16"
            height="16"
            fill="none"
            stroke="currentcolor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          >
            <path d="M30 12 L16 24 2 12" />
          </svg>}
    </IconContainer>
  );
};