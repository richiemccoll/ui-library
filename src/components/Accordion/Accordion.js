import React from "react";
import PropTypes from "prop-types";
import { Icons } from './Icons';
import { Container } from './UIElements';

class Accordion extends React.Component {
  constructor() {
    super();
    this.state = {
      isOpen: false
    };
  }

  expand = state => ({ isOpen: !state.isOpen });

  toggle() {
    this.setState(this.expand);
  }

  render() {
    const { isOpen } = this.state;
    const { title, children } = this.props;

    return (
      <Container>
        <li onClick={this.toggle.bind(this)}>
          {title}
          <Icons isOpen={isOpen} />
        </li>
        {isOpen && children}
      </Container>
    );
  }
}

export default Accordion;

Accordion.PropTypes = {
  title: PropTypes.string.isRequired
};
