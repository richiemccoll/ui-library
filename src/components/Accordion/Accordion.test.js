import React from "react";
import renderer from "react-test-renderer";
import Accordion from "./Accordion";

describe("Accordion Component", () => {
  it("should render correctly", () => {
    const component = renderer.create(<Accordion title="test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
