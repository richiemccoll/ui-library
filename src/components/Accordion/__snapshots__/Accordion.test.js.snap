// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component should render correctly 1`] = `
<ul
  className="sc-bdVaJa hwTUhO"
>
  <li
    onClick={[Function]}
  >
    test
    <span
      className="sc-bwzfXH jvACAx"
    >
      <svg
        fill="none"
        height="16"
        id="i-chevron-bottom"
        stroke="currentcolor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 32 32"
        width="16"
      >
        <path
          d="M30 12 L16 24 2 12"
        />
      </svg>
    </span>
  </li>
</ul>
`;
